src/main/java/frc/robot/subsystems/spinner/Spinner.java

```diff
@@ -0,0 +1,106 @@
+package frc.robot.subsystems.spinner;
+
+import com.ctre.phoenix.motorcontrol.NeutralMode;
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonSRX;
+import com.revrobotics.ColorMatch;
+import com.revrobotics.ColorMatchResult;
+import com.revrobotics.ColorSensorV3;
+import edu.wpi.first.wpilibj.util.Color;
+import frc.robot.constants.RobotMap;
+import frc.robot.subsystems.OverridableSubsystem;
+import io.github.oblarg.oblog.annotations.Log;
+import frc.robot.constants.RobotConstants.SpinnerConstants;
+
+public class Spinner extends OverridableSubsystem {
+  private final WPI_TalonSRX talonSRX;
+  private final ColorSensorV3 colorSensor;
+  private final ColorMatch colorMatcher;
+  private Color kBlue;
+  private Color kGreen;
+  private Color kRed;
+  private Color kYellow;
+
+  /**
+   * This class holds all of the methods for the trench roller subsystem, which
+   * spins the trench.
+   */
+  public Spinner() {
+    talonSRX = new WPI_TalonSRX(RobotMap.kRollerTalonSRX);
+    talonSRX.setNeutralMode(NeutralMode.Coast);
+    talonSRX.setInverted(SpinnerConstants.kIsInverted);
+    talonSRX.configOpenloopRamp(SpinnerConstants.kRampRate);
+
+    colorSensor = new ColorSensorV3(RobotMap.kI2cPort);
+    colorMatcher = new ColorMatch();
+
+    createColors();
+  }
+
+  @Override
+  public void overriddenMove(double power) {
+    talonSRX.set(power);
+  }
+
+  /**
+   * @return an enum of the color, including unknown if the minimum threshold is
+   *         not met.
+   */
+  public Color getColor() {
+    return colorSensor.getColor();
+  }
+
+  /**
+   * Compare two colors
+   * 
+   * @return are the two colors equal.
+   */
+  public boolean compareColors(Color color, Color comparedColor) {
+    ColorMatchResult match = colorMatcher.matchClosestColor(comparedColor);
+    return match.color == comparedColor;
+    //return color == comparedColor;
+  }
```

Comment:
`public boolean compareColors(Color color, Color colorToCompare) {
  ColorMatchResult match = colorMatcher.matchClosestColor(color);
  return match.color == colorToCompare;
}`

Reply:
nice review i wish you would do more of these